checkov/arm/graph_builder/local_graph.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+import logging
+from copy import deepcopy
+from typing import Any, TYPE_CHECKING
+
+from checkov.arm.graph_builder.graph_components.block_types import BlockType
+from checkov.arm.graph_builder.graph_components.blocks import ArmBlock
+from checkov.arm.utils import ArmElements
+from checkov.common.graph.graph_builder import CustomAttributes
+from checkov.common.graph.graph_builder.graph_components.blocks import Block
+from checkov.common.graph.graph_builder.local_graph import LocalGraph
+from checkov.common.util.consts import START_LINE, END_LINE
+
+if TYPE_CHECKING:
+    from checkov.common.graph.graph_builder.local_graph import _Block
+
+
+class ArmLocalGraph(LocalGraph[Block]):
+    def __init__(self, definitions: dict[str, dict[str, Any]]) -> None:
+        super().__init__()
+        self.vertices: list[Block] = []
+        self.definitions = definitions
+        self.vertices_by_path_and_name: dict[tuple[str, str], int] = {}
+
+    def build_graph(self, render_variables: bool = False) -> None:
+        self._create_vertices()
+        logging.debug(f"[ArmLocalGraph] created {len(self.vertices)} vertices")
+
+        self._create_edges()
+        logging.debug(f"[ArmLocalGraph] created {len(self.edges)} edges")
+
+    def _create_vertices(self) -> None:
+        for file_path, definition in self.definitions.items():
+            self._create_parameter_vertices(file_path=file_path, parameters=definition.get(ArmElements.PARAMETERS))
+            self._create_resource_vertices(file_path=file_path, resources=definition.get(ArmElements.RESOURCES))
+
+        for i, vertex in enumerate(self.vertices):
+            self.vertices_by_block_type[vertex.block_type].append(i)
+            self.vertices_block_name_map[vertex.block_type][vertex.name].append(i)
+            self.vertices_by_path_and_name[(vertex.path, vertex.name)] = i
```

Comment:
is `vertex.name` unique? I think you should use `vertex.id`. 
you can have resources from different types with the same name

Reply:
yeah, sounds about right 😄